test/Interop/Cxx/templates/template-type-parameter-not-in-signature.swift

```diff
@@ -0,0 +1,22 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop -Xfrontend -validate-tbd-against-ir=none)
+//
+// REQUIRES: executable_test
+
+import TemplateTypeParameterNotInSignature
+import StdlibUnittest
+
+var TemplateNotInSignatureTestSuite = TestSuite("Template Type Parameters Not in Function Signature")
+
+TemplateNotInSignatureTestSuite.test("Function with defaulted template type parameters") {
+  templateTypeParamNotUsedInSignature(T: Int.self)
+  multiTemplateTypeParamNotUsedInSignature(T: Float.self, U: Int.self)
+  multiTemplateTypeParamNotUsedInSignatureWithUnrelatedParams(1, 1, T: Int32.self, U: Int.self)
+  let _: Int = templateTypeParamUsedInReturnType(10)
```

Comment:
Can you add an `expectEquals` somewhere in both of these tests. Otherwise (very theoretically), they might just get optimized away. Also it's not a bad idea to just test that the correct thing is returned and there wasn't an argument mismatch or something.